src/protagonist/CleanupHandler/AssetUpdatedHandler.cs

```diff
@@ -0,0 +1,357 @@
+
+        RemoveObjectsFromFolderInBucket(infoJsonRoot);
+    }
+
+    private async Task CleanupModified(List<ImageDeliveryChannel> modifiedOrAdded, Asset assetBefore, Asset assetAfter)
+    {
+        foreach (var deliveryChannel in modifiedOrAdded)
+        {
+            if (assetBefore.ImageDeliveryChannels.Any(x => x.Channel == deliveryChannel.Channel)) // checks for updated rather than added
+            {
+                await CleanupChangedPolicy(deliveryChannel, assetAfter);
+            }
+        }
+    }
+    
+    private async Task CleanupRemoved(ImageDeliveryChannel deliveryChannelRemoved, Asset assetAfter)
+    {
+        switch (deliveryChannelRemoved.Channel)
+        {
+            case AssetDeliveryChannels.Image:
+                CleanupRemovedImageDeliveryChannel(assetAfter);
+                break;
+            case AssetDeliveryChannels.Thumbnails:
+                await CleanupRemovedThumbnailDeliveryChannel(assetAfter);
+                break;
+            case AssetDeliveryChannels.Timebased:
+                await CleanupRemovedTimebasedDeliveryChannel(assetAfter);
+                break;
+            case AssetDeliveryChannels.File:
+                CleanupFileDeliveryChannel(assetAfter);
+                break;
+            default:
+                logger.LogDebug("policy {PolicyName} does not require any changes for asset {AssetId}",
+                    deliveryChannelRemoved.DeliveryChannelPolicy.Name, assetAfter.Id);
+                break;
+        }
+    }
+
+    private async Task CleanupChangedPolicy(ImageDeliveryChannel deliveryChannelModified, Asset assetAfter)
+    {
+        switch (deliveryChannelModified.Channel)
+        {
+            case AssetDeliveryChannels.Image:
+                CleanupChangedImageDeliveryChannel(deliveryChannelModified, assetAfter.Id);
+                break;
+            case AssetDeliveryChannels.Thumbnails:
+                await CleanupChangedThumbnailDeliveryChannel(assetAfter);
+                break;
+            case AssetDeliveryChannels.Timebased:
+                await CleanupChangedTimebasedDeliveryChannel(deliveryChannelModified, assetAfter.Id);
+                break;
+            default:
+                logger.LogDebug("policy {PolicyName} does not require any changes for asset {AssetId}",
+                    deliveryChannelModified.DeliveryChannelPolicy.Name, assetAfter.Id);
+                break;
+        }
+    }
+
+    private async Task CleanupChangedTimebasedDeliveryChannel(ImageDeliveryChannel imageDeliveryChannel, AssetId assetId)
+    {
+        var presetList = JsonSerializer.Deserialize<List<string>>(imageDeliveryChannel.DeliveryChannelPolicy.PolicyData);
+        List<ObjectInBucket> assetsToDelete;
+        var keys = new List<string>();
+        var containers = new List<string>();
+
+        foreach (var presetIdentifier in presetList)
+        {
+            var presetDictionary = await engineClient.GetAvPresets();
```

Comment:
This can be called once, don't need to call per preset. 

The results of this call should also be cached to avoid repeated HTTP requests to Engine for data that's unlikely to change. What happens if we can't fetch presets from Engine? The changes are it'll be due to a transient failure so maybe worth failing the message processing and picking up later?